src/backend/utils/cache/inval.c

```diff
@@ -1326,6 +1345,127 @@ CacheInvalidateHeapTuple(Relation relation,
 	RegisterRelcacheInvalidation(databaseId, relationId);
 }
 
+void
+CacheInvalidateENRHeapTuple(Relation relation,
```

Comment:
This looks like almost a 1:1 copy of CacheInvalidateHeapTuple - is it not possible to just modify CacheInvalideHeapTuple rather than duplicating so much of its code? If not, we should have a comment to explain what the differences are, and why they are necessary

Reply:
It's not possible to modify CacheInvalidateHeapTuple because there is not enough information to determine whether the catalog in question is referencing an ENR catalog or not. Refactoring the function isn't desirable either because it would mean far more engine changes.